shared/metrics/src/main/java/io/pravega/shared/MetricsTags.java

```diff
@@ -0,0 +1,120 @@
+
+import com.google.common.base.Preconditions;
+
+public class MetricsTags {
+
+    // Metric Tag Names
+    public static final String TAG_CONTAINER = "container";
+    public static final String TAG_HOST = "host";
+    public static final String TAG_SCOPE = "scope";
+    public static final String TAG_STREAM = "stream";
+    public static final String TAG_SEGMENT = "segment";
+    public static final String TAG_TRANSACTION = "transaction";
+    public static final String TAG_EPOCH = "epoch";
+
+    private static final String TRANSACTION_DELIMITER = "#transaction.";
+    private static final String EPOCH_DELIMITER = ".#epoch.";
+    /**
+     * The Transaction unique identifier is made of two parts, each having a length of 16 bytes (64 bits in Hex).
+     */
+    private static final int TRANSACTION_PART_LENGTH = Long.BYTES * 8 / 4;
+    /**
+     * The length of the Transaction unique identifier, in bytes (it is made of two parts).
+     */
+    private static final int TRANSACTION_ID_LENGTH = 2 * TRANSACTION_PART_LENGTH;
+
+    /**
+     * Generate string array as container tag for metric.
+     * @param containerId container id.
+     * @return string array as the container tag of metric.
+     */
+    public static String[] containerTag(int containerId) {
+        return new String[] {TAG_CONTAINER, String.valueOf(containerId)};
+    }
+
+    /**
+     * Generate string array as host tag for metric.
+     * @param hostname hostname of the metric.
+     * @return string array as the host tag of metric.
+     */
+    public static String[] hostTag(String hostname) {
+        return new String[] {TAG_HOST, hostname};
+    }
+
+    /**
+     * Generate string array as stream tags for metric.
+     * @param scope scope of the stream.
+     * @param stream stream name.
+     * @return string array as the stream tag of metric.
+     */
+    public static String[] streamTags(String scope, String stream) {
+        return new String[] {TAG_SCOPE, scope, TAG_STREAM, stream};
+    }
+
+    /**
+     * Generate string array as transaction tags for metric.
+     * @param scope scope of the stream.
+     * @param stream stream name.
+     * @param transactionId transaction id.
+     * @return string array as transaction tag of metric.
+     */
+    public static String[] transactionTags(String scope, String stream, String transactionId) {
+        return new String[] {TAG_SCOPE, scope, TAG_STREAM, stream, TAG_TRANSACTION, transactionId};
+    }
+
+    /**
+     * Generate string array as segment tags for metric.
+     * @param qualifiedSegmentName fully qualified segment name.
+     * @return string array as segment tag of metric.
+     */
+    public static String[] segmentTags(String qualifiedSegmentName) {
+        Preconditions.checkNotNull(qualifiedSegmentName);
+        String segmentBaseName = getSegmentBaseName(qualifiedSegmentName);
+
+        String[] tags = {TAG_SCOPE, null, TAG_STREAM, null, TAG_SEGMENT, null, TAG_EPOCH, null};
```

Comment:
You may want to add more inline comments of the actual inputs you expect to parse here. Also, a good testing of the different expected (and unexpected) inputs for this method is important.

Reply:
Yes many test cases created for metric tags and a new MetricsTags utility class generated.